chaste_codegen/tests/test_lookup_tables.py

```diff
@@ -1,113 +1,184 @@
 
-    for eq in hn_model.equations:
+    for eq in s_model.equations:
         with lut.method_being_printed('method2'):
             printer.doprint(eq.rhs)
+    print(lut.print_lookup_parameters(printer))
     assert lut.print_lookup_parameters(printer) == \
         [{'metadata_tag': 'membrane_voltage', 'mTableMins': -250.0001, 'mTableMaxs': 549.9999, 'mTableSteps': 0.001,
-          'table_used_in_methods': {'method1', 'method2'}, 'var': 'membrane$V',
-          'lookup_epxrs': ['8000.0 * exp(-(66.0 + membrane$V) * 0.056)', '20.0 * exp(-(75.0 + membrane$V) * 0.125)',
-                           '2000.0 / (320.0 * exp(-(75.0 + membrane$V) * 0.1) + 1.0)',
-                           'exp((-40.0 + membrane$V) * 0.08)']}]
-
-
-def test_change_lookup_table():
-    model_folder = os.path.join(cg.DATA_DIR, 'tests', 'cellml', 'beeler_reuter_model_1977.cellml')
-    model = load_model(model_folder)
-
-    lut = LookupTables(model, lookup_params=[['membrane_voltage', -25.0001, 54.9999, 0.01],
-                                             ['cytosolic_calcium_concentration', 0.0, 50.0, 0.01],
-                                             ['unknown_tag', 0.0, 50.0, 0.01]])
-    lut.calc_lookup_tables(model.equations)
+          'table_used_in_methods': {'method2', 'method1'}, 'var': 'cell$V',
+          'lookup_epxrs':
+            ['(47.13 + cell$V) * 0.32 / ((-exp(-(47.13 + cell$V) * 0.1) + 1.0) * 1.0)',
+             '0.08 * exp(-cell$V / 11.0)',
+             '1.0 / (1.0 + exp((-50.0 - cell$V) / 7.5))',
+             ('1.0 / ((10.0 + cell$V) * 0.00061 / (-1.0 + exp((10.0 + cell$V) * 0.145)) + (7.0 + cell$V) * 0.00138 / '
+              '(-exp(-(7.0 + cell$V) * 0.123) + 1.0))'),
+             '1.0 / (1.0 + exp((33.0 + cell$V) / 22.4))',
+             '1.0 / (1.0 + exp((-cell$V + 1.5) / 16.7))',
+             ('1.0 / ((30.0 + cell$V) * 0.000131 / (-1.0 + exp((30.0 + cell$V) * 0.0687)) + (30.0 + cell$V) * 7.19e-05'
+              ' / (-exp(-(30.0 + cell$V) * 0.148) + 1.0))'),
+             '1.0 / (1.0 + exp((-3.0 - cell$V) / 15.0))',
+             '9.0 / (1.0 + exp((3.0 + cell$V) / 15.0)) + 0.5',
+             '1.0 / (1.0 + exp((33.5 + cell$V) / 10.0))',
+             '3000.0 / (1.0 + exp((60.0 + cell$V) / 10.0)) + 30.0',
+             '1.0 / (1.0 + exp((33.5 + cell$V) / 10.0))',
+             '2800.0 / (1.0 + exp((60.0 + cell$V) / 10.0)) + 220.0',
+             '1.0 / (1.0 + exp((-3.0 - cell$V) / 15.0))',
+             '3.5 * exp(-pow((cell$V / 30.0), 2.0)) + 1.5',
+             '1.0 / (1.0 + exp((33.5 + cell$V) / 10.0))',
+             '20.0 / (1.0 + exp((33.5 + cell$V) / 10.0)) + 20.0',
+             '1.0 / (1.0 + exp((-14.5 - cell$V) / 6.0))',
+             '-exp((-14.5 - cell$V) / 6.0) + 1.0',
+             '1.0 / (1.0 + exp((35.06 + cell$V) / 3.6)) + 0.6 / (1.0 + exp((-cell$V + 50.0) / 20.0))',
+             '1.0 / (0.0197 * exp(-pow(((14.5 + cell$V) * 0.0337), 2.0)) + 0.02)']}]
+
+
+def test_change_lookup_table(be_model):
```

Comment:
Could all do with a lot more comments? If these tests stop working in 5 years will you remember what they were supposed to do?

Reply:
Most of these I think so as I've tried to at least give them descriptive names.